src/block/Sapling.php

```diff
@@ -100,23 +100,23 @@ public function ticksRandomly() : bool{
 	public function onRandomTick() : void{
 		if($this->position->getWorld()->getFullLightAt($this->position->getFloorX(), $this->position->getFloorY(), $this->position->getFloorZ()) >= 8 and mt_rand(1, 7) === 1){
 			if($this->ready){
-				$this->grow();
+				$this->grow(null, null);
 			}else{
 				$this->ready = true;
 				$this->position->getWorld()->setBlock($this->position, $this);
 			}
 		}
 	}
 
-	private function grow() : void{
+	private function grow(?Item $item, ?Player $player) : void{
```

Comment:
Any reason for not defaulting these parameters to null? Personally, passing null in the call can be a bit unsightly.

Reply:
I had read once that it was better to avoid the default parameters because they could be forgotten. Since the function is private I might as well use them. I await a response from @dktapps.

Reply:
Yes, optional parameters should be avoided in almost all cases.